pkg/crd/crd.go

```diff
@@ -344,17 +264,104 @@ func ConvertToPortSchema(crd v1.CustomResourceDefinition) ([]port.Action, *port.
 	}
 
 	return actions, &bp, nil
 }
+
+func findMatchingCRDs(crds []v1.CustomResourceDefinition, pattern string) []v1.CustomResourceDefinition {
+	matchedCRDs := make([]v1.CustomResourceDefinition, 0)
+
+	for _, crd := range crds {
+		mapCrd, err := goutils.StructToMap(crd)
+
+		if err != nil {
+			klog.Errorf("Error converting CRD to map: %s", err.Error())
+			continue
+		}
+
+		match, err := jq.ParseBool(pattern, mapCrd)
+
+		if err != nil {
+			klog.Errorf("Error running jq on crd CRD: %s", err.Error())
+			continue
+		}
+		if match {
+			matchedCRDs = append(matchedCRDs, crd)
+		}
+	}
+
+	return matchedCRDs
+}
+
+func handleMatchingCRD(crds []v1.CustomResourceDefinition, portConfig *port.IntegrationAppConfig, portClient *cli.PortClient) {
+	matchedCRDs := findMatchingCRDs(crds, portConfig.CRDSToDiscover)
+
+	for _, crd := range matchedCRDs {
+		portConfig.Resources = append(portConfig.Resources, createKindConfigFromCRD(crd))
+		actions, bp, err := convertToPortSchema(crd)
+		if err != nil {
+			klog.Errorf("Error converting CRD to Port schemas: %s", err.Error())
+			continue
+		}
+
+		_, err = blueprint.NewBlueprint(portClient, *bp)
+
+		if err != nil && strings.Contains(err.Error(), "taken") {
+			klog.Infof("Blueprint already exists, patching blueprint properties")
+			_, err = blueprint.PatchBlueprint(portClient, port.Blueprint{Schema: bp.Schema, Identifier: bp.Identifier})
+			if err != nil {
+				klog.Errorf("Error patching blueprint: %s", err.Error())
+			}
+		}
+
+		if err != nil {
+			klog.Errorf("Error creating blueprint: %s", err.Error())
+		}
+
+		for _, act := range actions {
+			_, err = blueprint.NewBlueprintAction(portClient, bp.Identifier, act)
+			if err != nil {
+				if strings.Contains(err.Error(), "taken") {
+					if portConfig.OverwriteCRDsActions == true {
+						_, err = blueprint.UpdateBlueprintAction(portClient, bp.Identifier, act)
+						if err != nil {
+							klog.Errorf("Error updating blueprint action: %s", err.Error())
+						}
+					} else {
+						klog.Infof("Action already exists, if you wish to overwrite it, delete it first or provide the configuration overwriteCrdsActions: true, in the exporter configuration and resync")
+					}
+				} else {
+					klog.Errorf("Error creating blueprint action: %s", err.Error())
+				}
+			}
+		}
+	}
+}
+
+func AutodiscoverCRDsToActions(portConfig *port.IntegrationAppConfig, apiExtensionsClient apiextensions.ApiextensionsV1Interface, portClient *cli.PortClient) {
+	if portConfig.CRDSToDiscover == "" {
+		klog.Info("Discovering CRDs is disabled")
+		return
+	}
+
+	klog.Infof("Discovering CRDs/XRDs with pattern: %s", portConfig.CRDSToDiscover)
+	crds, err := apiExtensionsClient.CustomResourceDefinitions().List(context.Background(), metav1.ListOptions{})
+
+	if err != nil {
+		klog.Errorf("Error listing CRDs: %s", err.Error())
+		return
+	}
+
+	handleMatchingCRD(crds.Items, portConfig, portClient)
```

Comment:
I think you should call the functions `handleCRDs` as you supply all the crds and not only the ones that matching

Reply:
Makes sense, nice catch!